File: python/MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY/MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY/abstract_factory_master/MembresiaFactory.py
from MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY.whole_part.MembresiaVIP import MembresiaVIP
from MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY.whole_part.MembresiaSencilla import MembresiaSencilla
from .ProductoCineFactory import ProductoCineFactory
from ..master_slaves.GestorDatosCliente     import GestorDatosCliente
from ..master_slaves.GestorDatosBeneficios  import GestorDatosBeneficios

class MembresiaFactory(ProductoCineFactory):
        
    def crearProductoCineSencillo(self,nombres,apellidoPaterno,
                       apellidoMaterno,edad,tipoCliente):
        datosCliente = GestorDatosCliente().preparaDatosCliente(nombres, apellidoPaterno, apellidoMaterno, edad, tipoCliente)
        return MembresiaSencilla(datosCliente)
    
    def crearProductoCineVIP(self,nombres,apellidoPaterno,
                       apellidoMaterno,edad,tipoCliente,beneficios):
        datosCliente    = GestorDatosCliente().preparaDatosCliente(nombres, apellidoPaterno, apellidoMaterno, edad, tipoCliente)
        datosBeneficios = GestorDatosBeneficios().preparaDatosBeneficios(beneficios)
        return MembresiaVIP(datosCliente,datosBeneficios)

File: php/MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY/principal/MembresiaFactory.php
<?php

require './membresias/MembresiaSencilla.php';
require_once './membresias/MembresiaVIP.php';



/**
 * Description of MembresiaFactory
 *
 * @author giltenorio
 */
class MembresiaFactory implements ProductoCineFactory{
    //put your code here
    public $membresia;
    public function crearProductoCineSencillo($nombre, $apellidoP, $apellidoM, $edad, $tipoC, $numTarjeta, $fecha, $asiento, $dia, $mes, $anio, $pelicula) {
        $datosCliente = GestorDatosCliente::preparaDatosCliente($nombre, $apellidoP, $apellidoM, $edad, $tipoC);
        
        $membresia = new MembresiaSencilla($datosCliente,10,01,2022);
        return $membresia;
    }

    public function crearProductoCineVIP($nombre, $apellidoP, $apellidoM, $edad, $tipoC, $numTarjeta, $fecha, $asiento, $dia, $mes, $anio, $pelicula, $beneficio) {
            
        $datosCliente = GestorDatosCliente::preparaDatosCliente($nombre, $apellidoP, $apellidoM, $edad, $tipoC);
        
        $datosBeneficio = GestorDatosBeneficio::preparaDatosBeneficio($beneficio);
        
        $membresia = new MembresiaVIP($datosCliente, $datosBeneficio,10,01,2022);
        
        
        return $membresia;


    }

}
